MsCorePkg/Universal/StatusCodeHandler/Serial/Common/SerialStatusCodeHandler.c

```diff
@@ -158,7 +157,7 @@ SerialStatusCode(
   //
   // Call SerialPort Lib function to do print.
   //
-  WriteStatusCode((UINT8*)Buffer, CharCount);
+  WriteStatusCode((UINT8 *)BufferPtr, CharCount);
```

Comment:
If BufferPtr is still pointing at Data->String.Ascii, will this print the entire formatted string, or only part of the string?

Reply:
It will print CharCount chars. CharCount is provided by whomever generated the Data->String.Ascii and must be the correct number of chars to print. If someone submits a EFI_STATUS_CODE_STRING_DATA for printing where Data->String.Ascii and Data->DataHeader.Size are inconsistent, that's a bug in the creator of the EFI_STATUS_CODE_STRING_DATA.